lib/Habrok.js

```diff
@@ -78,13 +79,17 @@ Habrok.prototype.request = function request(raw, opts) {
     } else if (errorCode === ECONNRESET || statusCode > 299) {
       if ((!err && self.RETRY_CODES.indexOf(statusCode) === -1) || attempt >= self.RETRIES) {
         debug('fail', req);
+        debugRequest({ attempts: attempt });
```

Comment:
Just one suggest change. I would move all calls of debugRequest to the end as to ensure it is always called.

For example

```js
return resolve({
  statusCode: res.statusCode,
  headers: res.headers,
  body
})
.then((out) => {
  debugRequest({ attempts: attempt });
  return out;
})
.catch((err) => {
  debugRequest({ attempts: attempt });
  throw err;
});
```

Reply:
I think this is a good suggestion, but since the current way is working, and I couldn't figure out how to get this working, and we have an deadline coming up, I'm going to not change to this, if that is alright. 

Ready for your final review / merge if you are. And then we can publish to npm?

Reply:
(thumbsup)